app/models/user.rb

```diff
@@ -1,5 +1,13 @@
 class User < ApplicationRecord
-  has_many :posts, dependent: :destroy
-  has_many :likes, dependent: :destroy
-  has_many :comments, dependent: :destroy
+  has_many :posts, foreign_key: 'author_id'
+  has_many :comments, foreign_key: 'author_id'
+  has_many :likes, foreign_key: 'author_id'
+
+  def recent_posts
+    posts.order(created_at: :desc).limit(3)
+  end
+
+  def update_posts_count
+    update(posts_count: posts.count)
+  end
```

Comment:
- [ ] Reiterating the point from the previous reviewer, there is no need to have this method here. Simply place it in the `Post` model and ensure it runs automatically whenever a new post is created using the Rails `after_create` callback

This snippet should be helpful
```ruby
def update_posts_counter
  author.increment!(:posts_counter)
end
```